cmd/soroban-cli/src/output.rs

```diff
@@ -16,26 +16,44 @@ impl Output {
         Output { quiet }
     }
 
-    fn print<T: Display>(&self, icon: &str, message: T) {
-        if !self.quiet {
+    pub fn print<T: Display>(&self, icon: &str, message: T, new_line: bool) {
+        if self.quiet {
+            return;
+        }
+
+        if new_line {
             eprintln!("{icon} {message}");
+        } else {
+            eprint!("{icon} {message}");
```

Comment:
For consistency with the Rust stdlib, it'd be helpful if we operated the newline option via different function names, `print` and `println` like:
- https://doc.rust-lang.org/std/macro.print.html
- https://doc.rust-lang.org/std/macro.println.html

Other functions can then also get two variants, `check` and `checkln`.

Then I think we don't need the `icon` parameter to print and it can simply accept a `&str` (or preferably a `Display`). For example, the use would be:
```rust
output.globeln(format!("Downloading history {history_url}"));
...
output.bucket(format!("Downloading bucket {bucket_index} {bucket}…"));
...
output.println("({size})");
```

Reply:
I also wonder if we should use a macro to generate these functions, as they're essentially the same thing.